File: 57-insert-interval/insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        int n=intervals.size();
        vector<vector<int>>ans;
        int idx=n;
        for(int i=0;i<n;i++){
            //case-1
            if(intervals[i][1]<newInterval[0]){
                ans.push_back(intervals[i]);
            }else if(intervals[i][0]>newInterval[1]){//case-2
               idx=i;
               break;
            }else{
                newInterval[0]=min(newInterval[0],intervals[i][0]);
                newInterval[1]=max(newInterval[1],intervals[i][1]);
            }
        }
        ans.push_back(newInterval);
        for(int i=idx;i<n;i++){
            ans.push_back(intervals[i]);
        }
        return ans;
    }
};

File: 57-insert-interval/insert-interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        ArrayList<int []>ans=new ArrayList<>();
        int n=intervals.length;
        //store interval before overlap
        int idx=0;
        while(idx<n && newInterval[0] > intervals[idx][1] ){
            ans.add(intervals[idx]);
            idx++;
        }
        //merge overlapping intervals if any
        while(idx<n && newInterval[1]>=intervals[idx][0]){
          newInterval[0]=Math.min(newInterval[0],intervals[idx][0]);
          newInterval[1]=Math.max(newInterval[1],intervals[idx][1]);

          idx++;
        }
        ans.add(newInterval);

        while(idx<n){
            ans.add(intervals[idx]);
            idx++;
        }

        return ans.toArray(new int [ans.size()][2]);

       
    }
}